Bottom: 134e72593483ee8f0d72deeae6e09127b5cec8a6
Top:    694773a59a5b9c5020a67ac9f780d455ae9da338
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-22 17:03:36 +0800

Refresh of vhost_net-prefetch-desc

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index fcefa84..d5e3345 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -750,6 +750,7 @@ static void handle_rx_batched(struct vhost_net *net, struct vhost_log *vq_log)
 	struct socket *sock = vq->private_data;
 	unsigned int out, in, log = 0;
 	__virtio16 indices[VHOST_RX_BATCH];
+	int lens[VHOST_RX_BATCH];
 	int sock_len, i;
 	int avails, head;
 
@@ -766,10 +767,9 @@ static void handle_rx_batched(struct vhost_net *net, struct vhost_log *vq_log)
 			return;
 		}
 		for (i = 0; i < avails; i++) {
-			int len =
-				__skb_array_len_with_tag(nvq->rxq[nvq->rh + i]);
+			lens[i] = __skb_array_len_with_tag(nvq->rxq[nvq->rh + i]);
 			vhost_add_used_elem(vq, indices[i],
-					    cpu_to_vhost32(vq, len), i);
+					    cpu_to_vhost32(vq, lens[i]), i);
 		}
 		for (i = 0; i < avails; i++) {
 			if (nvq->rh == nvq->rt) {
@@ -784,7 +784,7 @@ static void handle_rx_batched(struct vhost_net *net, struct vhost_log *vq_log)
 				return;
 			}
 			if (rx_recvmsg(nvq, in, nvq->rxq[nvq->rh++],
-				       sock_len, vq_log, log)) {
+				       lens[i], vq_log, log)) {
 				printk("recvmsg error!\n");
 				return;
 			}
